pkg/pool-weighted/contracts/managed/ManagedPoolSettings.sol

```diff
@@ -722,34 +711,31 @@ abstract contract ManagedPoolSettings is BasePool, ProtocolFeeCache, IControlled
         uint256 mintAmount,
         address recipient
     ) external authenticate whenNotPaused {
-        // This complex operation might mint BPT, altering the supply. For simplicity, we forbid adding tokens before
-        // initialization (i.e. before BPT is first minted). We must also collect AUM fees every time the BPT supply
-        // changes. For consistency, we do this always, even if the amount to mint is zero.
-        uint256 supply = _getVirtualSupply();
-        _require(supply > 0, Errors.UNINITIALIZED);
-        _collectAumManagementFees(supply);
-
-        // BPT cannot be added using this mechanism: Composable Pools manage it via dedicated PoolRegistrationLib
-        // functions.
-        _require(tokenToAdd != IERC20(this), Errors.ADD_OR_REMOVE_BPT);
-
-        // Tokens cannot be added during or before a weight change, since a) adding a token already involves a weight
-        // change and would override an existing one, and b) any previous weight changes would be incomplete since they
-        // wouldn't include the new token.
-        _ensureNoWeightChange();
-
-        // We first register the token in the Vault. This makes the Pool enter an invalid state, since one of its tokens
-        // has a balance of zero (making the invariant also zero). The Asset Manager must be used to deposit some
-        // initial balance and restore regular operation.
-        //
-        // We don't need to check that the new token is not already in the Pool, as the Vault will simply revert if we
-        // try to register it again.
-        PoolRegistrationLib.registerToken(getVault(), getPoolId(), tokenToAdd, assetManager);
-
-        // With the token registered, we fetch the new list of Pool tokens (which will include it). This is also a good
-        // opportunity to check we have not added too many tokens.
+        {
+            // This complex operation might mint BPT, altering the supply. For simplicity, we forbid adding tokens
+            // before initialization (i.e. before BPT is first minted). We must also collect AUM fees every time the
+            // BPT supply changes. For consistency, we do this always, even if the amount to mint is zero.
+            uint256 supply = _getVirtualSupply();
+            _require(supply > 0, Errors.UNINITIALIZED);
+            _collectAumManagementFees(supply);
+        }
```

Comment:
Could move this into a function for both add and remove, save bytecode and not need the scope:

```
   /**
     * @dev Complex Add/Remove operations might mint or burn BPT, altering the supply. For simplicity, we forbid
     * adding or removing tokens before initialization (i.e. before BPT is first minted). We must also collect
     * AUM fees every time the BPT supply changes. For consistency, we do this always, even if the amount to
     * mint is zero.
     * This should be called on all adds/removes.
     */
    function _ensureAumManagementFeesCollected() private {
        uint256 supply = _getVirtualSupply();
        _require(supply > 0, Errors.UNINITIALIZED);
        _collectAumManagementFees(supply);
    }
```

Reply:
The savings would not be large - I'd rather keep this explicit for now